boot/freeldr/freeldr/ntldr/wlregistry.c

```diff
@@ -477,6 +480,16 @@ WinLdrLoadNLSData(IN OUT PLOADER_PARAMETER_BLOCK LoaderBlock,
     return FALSE;
 }
 
+typedef struct {
+    LIST_ENTRY ListEntry;
+    ULONG Group;
+    ULONG Tag;
+    ULONG ErrorControl;
+    ULONG NameLength;
+    ULONG ImagePathLength;
+    BOOLEAN Processed;
+} DriverRegistryEntry;
```

Comment:
Please stick to out coding style.

Reply:
Comment 1 - I'm hard pressed to answer that, because as you say the original code doesn't seem to make any sense. Vista HAL will crash if page 0 isn't identity-mapped (it looks like it copies the real-mode IVT somewhere), but that won't have any effect without this bugfix.

Comment 3 - sorry, I don't know what you mean. Is it a problem with the struct itself? Or should it be at the top of the file? Or do you need it to be in a header file?

Reply:
That's the coding style (standard NT):
```c
typedef struct _DRIVER_REGISTRY_ENTRY
{
    MY_TYPE Member;
    // etc...
} DRIVER_REGISTRY_ENTRY, *PDRIVER_REGISTRY_ENTRY;
```

Reply:
Fair enough. Thanks